pkg/dynconf/dyn_settings.go

```diff
@@ -351,5 +389,24 @@ func (ds *DynSettings) GetTLSConfigFromPath(certificatesPath string) *tls.Config
 
 // Returns true if no value has been set in either the context, config file or command line for the supplied flag name
 func (ds *DynSettings) IsUsingDefaultValue(name string) bool {
-	return ds.defaultValueIsUsed[name]
+	_, ok := ds.defaultParams[name]
```

Comment:
This does not work anymore since the code building the defaultParams array above only takes commandline parameters into account. Maybe modify this to check if the parameter is in configParams or contextParams or commandLineParams.